File: ParkPal.Application/src/models/api/requests/notification/EditNotificationRequest.ts
export default class EditNotificationRequest {
    notificationId?: number;
    criteriaType?: number;
    waitTime?: number;

    constructor(data: Pick<EditNotificationRequest, "notificationId" | "criteriaType" | "waitTime"> | null = null) {
        if(data != null) {
            this.notificationId = data.notificationId;
            this.criteriaType = data.criteriaType;
            this.waitTime = data.waitTime;
        }
    }
}

File: ParkPal.API/ParkPal.API/Models/Requests/Notification/EditNotificationRequest.cs
using ParkPal.Common.Models.Database.Entities.Notification.Enums;

namespace ParkPal.API.Models.Requests.Notification;

public class EditNotificationRequest
{
    public int NotificationId { get; set; }
    public CriteriaType CriteriaType { get; set; }
    public int WaitTime { get; set; }
}